File: software/examples/vbnet/ExampleSimple.vb
Imports System
Imports Tinkerforge

Module ExampleSimple
    Const HOST As String = "localhost"
    Const PORT As Integer = 4223
    Const UID As String = "XYZ" ' Change XYZ to the UID of your CO2 Bricklet 2.0

    Sub Main()
        Dim ipcon As New IPConnection() ' Create IP connection
        Dim co2 As New BrickletCO2V2(UID, ipcon) ' Create device object

        ipcon.Connect(HOST, PORT) ' Connect to brickd
        ' Don't use device before ipcon is connected

        ' Get current all values
        Dim co2Concentration, humidity As Integer
        Dim temperature As Short

        co2.GetAllValues(co2Concentration, temperature, humidity)

        Console.WriteLine("CO2 Concentration: " + co2Concentration.ToString() + " ppm")
        Console.WriteLine("Temperature: " + (temperature/100.0).ToString() + " °C")
        Console.WriteLine("Humidity: " + (humidity/100.0).ToString() + " %RH")

        Console.WriteLine("Press key to exit")
        Console.ReadLine()
        ipcon.Disconnect()
    End Sub
End Module

File: software/examples/go/example_simple.go
package main

import (
	"fmt"
	"github.com/Tinkerforge/go-api-bindings/co2_v2_bricklet"
	"github.com/Tinkerforge/go-api-bindings/ipconnection"
)

const ADDR string = "localhost:4223"
const UID string = "XYZ" // Change XYZ to the UID of your CO2 Bricklet 2.0.

func main() {
	ipcon := ipconnection.New()
	defer ipcon.Close()
	co2, _ := co2_v2_bricklet.New(UID, &ipcon) // Create device object.

	ipcon.Connect(ADDR) // Connect to brickd.
	defer ipcon.Disconnect()
	// Don't use device before ipcon is connected.

	// Get current all values.
	co2Concentration, temperature, humidity, _ := co2.GetAllValues()

	fmt.Printf("CO2 Concentration:  ppm\n", co2Concentration)
	fmt.Printf("Temperature: %f °C\n", float64(temperature)/100.0)
	fmt.Printf("Humidity: %f %%RH\n", float64(humidity)/100.0)

	fmt.Print("Press enter to exit.")
	fmt.Scanln()
}
